docs/first-build.rst

```diff
@@ -1,130 +1,96 @@
-Building your first Flatpak
+Building Your First Flatpak
 ===========================
 
-This tutorial provides a quick introduction to building Flatpaks. In it, you will learn how to create a basic Flatpak application, which can be installed and run.
+This tutorial provides a quick, step-by-step example of how to build a simple application as a Flatpak. It also takes the opportunity to explain a few key concepts.
 
-Please follow the :doc:`setup` page before doing this tutorial.
+In order to complete this tutorial, you should have followed the `setup guide on flatpak.org <http://flatpak.org/setup/>`_. You also need to have installed ``flatpak-builder``, which is usually available from the same repository as the ``flatpak`` package.
 
-1. Install a runtime and the matching SDK
------------------------------------------
+1. Create a manifest
+--------------------
 
-Flatpak requires every app to specify a runtime that it uses for its basic
-dependencies. Each runtime has a matching SDK (Software Development Kit), which
-contains all the things that are in the runtime, plus headers and development
-tools. This SDK is required to build apps for the runtime.
-
-In this tutorial we will use the Freedesktop 1.6 runtime and SDK. To install these, run::
-
-  $ flatpak install flathub org.freedesktop.Platform//1.6 org.freedesktop.Sdk//1.6
-
-2. Create the app
------------------
-
-The app that is going to be created for this tutorial is a simple script. To
-create it, copy the following::
-
-  #!/bin/sh
-  echo "Hello world, from a sandbox"
-
-Now paste this into an empty file and save it as `hello.sh`.
-
-3. Add a manifest
------------------
-
-Each Flatpak is built using a manifest file which provides basic information about the application and instructions for how it is to be built. To add a manifest to the hello world app, add the following to an empty file:
-
-.. code-block:: json
+The input to ``flatpak-builder`` is a JSON file that describes the parameters for building the application. This is called the manifest. The following example is the manifest for the GNOME Dictionary application::
 
   {
-      "app-id": "org.flatpak.Hello",
-      "runtime": "org.freedesktop.Platform",
-      "runtime-version": "1.6",
-      "sdk": "org.freedesktop.Sdk",
-      "command": "hello.sh",
-      "modules": [
+    "app-id": "org.gnome.Dictionary",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "command": "gnome-dictionary",
+    "finish-args": [
+       "--socket=x11",
```

Comment:
I realize you are trying to be minimal but this leaves some bad habits like not including `--socket=wayland` and `--share=ipc` for Gtk3 applications.

Reply:
The only reason those are the only sandbox permissions in the example is that they're what's in the manifest I inherited. I'd love to have a better demo with a sane set of options in it (that's #97)!

Reply:
All Gtk3 applications will have like 7 or so permissions to just be functional.